db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,2200 @@
+     * The rules are as follows. Refer to the docs for more details on why they are what they are:
+     *    1. An External Event superseded by nothing will be present in the final, derived result.
+     *    2. An External Event partially superseded by a later External Source, but whose start time occurs before
+     *          the start of said External Source(s), will be present in the final, derived result.
+     *    3. An External Event whose start is superseded by another External Source, even if its end occurs after the
+     *          end of said External Source, will be replaced by the contents of that External Source (whether they are
+     *          blank spaces, or other events).
+     *    4. An External Event who shares a key with an External Event in a later External Source will always be
+     *          replaced.
+     */
+    @Nested
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    class DerivedEventRuleTests {
+
+      // Commonly Repeated:
+      @BeforeEach
+      void beforeEach() throws SQLException {
+        // insert generic external event type, source type, and derivation group
+        insertStandardTypes();
+      }
+
+      ////////////////////////// RULE 1 //////////////////////////
+      /**
+       * A solitary event shouldn't be superseded by anything.
+       *
+       *    A:   aa
+       */
+      @Test
+      void rule1_solitary() throws SQLException {
+        // insert the event(s) (and their source(s))
+        ExternalSource eS = new ExternalSource(
+            "A",
+            SOURCE_TYPE,
+            DERIVATION_GROUP,
+            "2024-01-01T00:00:00Z",
+            "2024-01-01T00:00:00Z",
+            "2024-01-01T01:00:00Z",
+            CREATED_AT
+        );
+
+        ExternalEvent e = new ExternalEvent("A.1", "2024-01-01T00:00:00Z", "01:00:00", eS);
+
+
+        // insert sources
+        insertExternalSource(eS);
+
+        // insert events
+        insertExternalEvent(e);
+
+        // ensure the result has the right size and keys
+        final var results = getDerivedEvents();
+
+        // both ranges should only have a single element and be fully present
+        final List<DerivedEvent> expectedResults = List.of(
+            new DerivedEvent(
+                "A.1",
+                SOURCE_TYPE,
+                "A",
+                DERIVATION_GROUP,
+                "2024-01-01 00:00:00+00",
+                "01:00:00",
+                "{[\"2024-01-01 00:00:00+00\",\"2024-01-01 01:00:00+00\")}",
+                "2024-01-01 00:00:00+00"
+            )
+        );
+        assertEquals(expectedResults.size(), results.size());
+        assertTrue(results.containsAll(expectedResults));
+      }
+
+      /**
+       * This test more thoroughly examines the lack of supercession with other sources. The event is bookended, but
+       *    none of the surrounding sources and their events can/should overwrite it.
```

Comment:
Again, rewording to be clearer:

```suggestion
       * An event that is in a partially overlapped source but outside of the overlap is not superseded.
```